apps/rig_kafka/lib/rig_kafka/client.ex

```diff
@@ -76,30 +76,37 @@ defmodule RigKafka.Client do
 
       content_type = get_content_type(headers_no_prefix, body)
 
-      decoded_body =
-        cond do
-          content_type == "avro/binary" ->
-            data =
+      try do
+        decoded_body =
+          case content_type do
+            "avro/binary" ->
+              data =
+                body
+                |> Serializer.decode_body!("avro", schema_registry_host: schema_registry_host)
+                |> Jason.decode!()
+
+              Map.merge(headers_no_prefix, %{data: data})
+
+            "application/json" ->
               body
-              |> Serializer.decode_body("avro", schema_registry_host)
-              |> Jason.decode!()
 
-            Map.merge(headers_no_prefix, %{data: data})
+            _ ->
+              body
```

Comment:
> > Integration tests fail on my machine. Tried multiple runs; this came up every time:
> > ```
> >   1) test Given a started RigKafka client, messages can be produced and consumed. (RigKafkaTest)
> >      test/rig_kafka_test.exs:42
> >      No message matching :test_message_received after 10000ms.
> >      The process mailbox is empty.
> >      code: assert_receive :test_message_received, 10_000
> >      stacktrace:
> >        test/rig_kafka_test.exs:62: (test)
> > ```
> > 
> > 
> > and this only during the first run:
> > ```
> >  1) test Given response_from is set to Kafka, the http response is taken from the Kafka response topic instead of forwarding the backend's original response. (RigTests.Proxy.ResponseFrom.KafkaTest)
> >      test/proxy/response_from/kafka_test.exs:47
> >      ** (HTTPoison.Error) :timeout
> >      code: %HTTPoison.Response{status_code: res_status, body: res_body} = HTTPoison.get!(request_url)
> >      stacktrace:
> >        (httpoison) lib/httpoison.ex:128: HTTPoison.request!/5
> >        test/proxy/response_from/kafka_test.exs:109: (test)
> > ```
> 
> First one fails even on master, it's tracked in #201. Second one can't confirm, but maybe caused by the same thing as first one.

Actually probably found the issue in our old test and made the tests more stable.